templates/release-docs-changelog.yml

```diff
@@ -0,0 +1,59 @@
+spec:
+  inputs:
+    stage:
+      description: The pipeline stage where to perform the docs release
+      default: publish
+    runner:
+      description: GitLab CI runner tag
+      default: hetzner-amd-beefy
+    alpine-git-tag:
+      description: Tag for alpine/git container image
+      default: latest
+    changelog_file:
+      description: Remote changelog file
+      default: CHANGELOG.md
+---
+
+
+release:mender-docs-changelog:
+  stage: $[[ inputs.stage ]]
+  tags:
+    - $[[ inputs.runner ]]
+  needs: []
+  image:
+    name: alpine/git:$[[ inputs.alpine-git-tag ]]
+    entrypoint: [""]
+  rules:
+    # Only make available for stable branches
+    - if: '$CI_COMMIT_TAG =~ /^v?\d+\.\d+\.\d+$/'
+      allow_failure: true
+  before_script:
+    # Check required environment variables
+    - |
+      if [ -z "$GITHUB_USER_EMAIL" ] || [ -z "$GITHUB_USER_NAME" ] || [ -z "$GITHUB_CLI_TOKEN" ] || [ -z "$GITHUB_BOT_TOKEN_REPO_FULL" ]; then
+        echo "ERROR: Missing required environment variables:"
+        echo "  - GITHUB_USER_EMAIL: ${GITHUB_USER_EMAIL:-'NOT SET'}"
+        echo "  - GITHUB_USER_NAME: ${GITHUB_USER_NAME:-'NOT SET'}"
+        echo "  - GITHUB_CLI_TOKEN: ${GITHUB_CLI_TOKEN:-'NOT SET'}"
+        echo "  - GITHUB_BOT_TOKEN_REPO_FULL: $([ -z "$GITHUB_BOT_TOKEN_REPO_FULL" ] && echo 'NOT SET' || echo 'SET')"
+        exit 1
+      fi
+    # Install required tools
+    - apk add --no-cache python3 curl
+    - curl -fsSL https://cli.github.com/packages/alpine/install.sh | sh
+    # Setting up git
+    - git config --global user.email "${GITHUB_USER_EMAIL}"
+    - git config --global user.name "${GITHUB_USER_NAME}"
+    # GITHUB_TOKEN for Github cli authentication
+    - export GITHUB_TOKEN=${GITHUB_CLI_TOKEN}
+  script:
+    - git clone https://${GITHUB_USER_NAME}:${GITHUB_BOT_TOKEN_REPO_FULL}@github.com/${GITHUB_CHANGELOG_REPO_URL}
+    - cd ${GITHUB_CHANGELOG_REPO_URL#*/}
+    - git checkout -b changelog-${CI_JOB_ID}
+    - [[ -f ../docs_header.md ]] && cat ../.docs_header.md > $[[ inputs.changelog_file ]]
+    - [[ -f ../CHANGELOG.md ]] && cat ../CHANGELOG.md | grep -v -E '^---' >> $[[ inputs.changelog_file ]]
```

Comment:
I see that we added now the checks. But if these files do not exist we should error. Also, I don't get why we use hard-coded ../CHANGELOG.md and $[[ inputs.changelog_file ]]. I don't think that it should be an input, actually, in my opinion it should be a hard requirement that `CHANGELOG.md` exists in the root of the repository. In which circumstances you think it makes sense to parametrize?

(Side note, I believe that the default shell in Alpine linux is not bash, so the built-in `[[` will fail. In general I recommend using `[` so that the script is porbable. Please try running the script on an Alipine Linux to verify)